dali/pipeline/pipeline.h

```diff
@@ -195,12 +194,14 @@ class DLL_PUBLIC Pipeline {
    * @param stream CUDA stream to use in case of GPUBackend
    * @param sync If SetExternalInputHelper should be blocking - waits until provided data is copied
    *             to the internal buffer
-   */
-  template<typename Backend>
-  DLL_PUBLIC inline void
-  SetExternalInput(const string &name, const TensorList<Backend> &tl, cudaStream_t stream = 0,
-                   bool sync = false, bool use_copy_kernel = false) {
-    SetExternalInputHelper(name, tl, stream, sync, use_copy_kernel);
+   * @param no_copy_mode Select whether to use the parameter defined in the External Source or
+   *                     override the mode of operation forcing the copy or no-copy
+   */
+  template <typename Backend>
+  DLL_PUBLIC inline void SetExternalInput(
+      const string &name, const TensorList<Backend> &tl, cudaStream_t stream = 0, bool sync = false,
+      bool use_copy_kernel = false, DALIExtNoCopyMode no_copy_mode = DALIExtNoCopyMode::DEFAULT) {
```

Comment:
Why not to use `SetDataConfig` in the `SetExternalInput` as well?

Reply:
Cause it's the most external part of the API, so I decided not to make a breaking change. If we find out that the list of additional flags is growing, we can make a break then.

I don't think I can make an overload with it as optional and those two would clash.

Reply:
OK